### PR TITLE
Update facades to support Juju 2.0 login

### PR DESCRIPTION
**Pull request by hatched**
When juju-core updated to 2.0 they updated many of the facades. This is the first branch of supporting multiple facades. We have to rely on the juju version because we don't get a list of supported facades until we have logged in.

### CODE REVIEW

File: jujugui/static/gui/src/app/views/utils.js [hunk 1]
```diff
@@ -1857,6 +1857,31 @@ YUI.add('juju-view-utils', function(Y) {
       charmName, services, counter += 1) : name;
   };
 
+  /**
+    Compare two semver version strings and return if one is
```
Comment (frankban) [inline jujugui/static/gui/src/app/views/utils.js]: Return -1 if...
Return 1 if...
etc?

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4397/
Test FAILed.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4398/
Test FAILed.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4399/
Test FAILed.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4401/
Test FAILed.

**Comment by frankban** [comment]
QA is ok, I was able to login in both new and old core versions.

File: jujugui/static/gui/src/app/views/utils.js (hunk 1 above)
Comment (hatched) [inline jujugui/static/gui/src/app/views/utils.js]: done

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4405/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4406/
Test PASSed.

File: Makefile [hunk 2]
```diff
@@ -380,7 +380,7 @@ bumpversion: deps
 	bin/bumpversion $(VPART)
 
 .PHONY: dist
-dist: clean-all gui test-deps collect-requirements
+dist: clean-all deps fast-babel gui test-deps collect-requirements
```
Comment (frankban) [inline Makefile]: Thank you!

File: jujugui/static/gui/src/app/store/env/go.js [hunk 3]
```diff
@@ -504,13 +507,29 @@ YUI.add('juju-env-go', function(Y) {
...
       if (credentials && credentials.areAvailable) {
+        var user = credentials.user;
+        var password = credentials.password;
+        var version = 0;
+        var params = {
+          AuthTag: credentials.user,
+          Password: credentials.password
+        };
+        // If the user is connecting to juju-core 2.0 or higher then we need
+        // to use the new params arguments. This is comparing against '2'
```
Comment (frankban) [inline jujugui/static/gui/src/app/store/env/go.js]: // to use the new params arguments and the version 2 of the facade.

**Comment by frankban** [comment]
:+1: I am QAing now.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4407/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4408/
Test PASSed.

**Comment by huwshimi** [comment]
This looks good? :+1:

**Comment by frankban** [comment]
QA ok

**Comment by hatched** [comment]
Thanks all! :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**